mesh.py

```diff
@@ -284,27 +284,44 @@ def __init__(self, meshParent, myID, equilibriumRegion, connections, radialIndex
         else:
             temp_psi_vals = self.psi_vals
 
-        # set sign of step in psi towards this region from primary separatrix
-        if temp_psi_vals[-1] - self.equilibriumRegion.psival > 0:
-            psi_sep_plus_delta = self.equilibriumRegion.psival + self.user_options.poloidal_spacing_delta_psi
-        else:
-            psi_sep_plus_delta = self.equilibriumRegion.psival - self.user_options.poloidal_spacing_delta_psi
 
         # Make vector along grad(psi) at start of equilibriumRegion
+        # Here we assume that the equilibriumRegion at a separatrix
+        # at the beginning and end, but not necessarily in between.
+        # This is to handle disconnected double null
+        start_point = self.equilibriumRegion[self.equilibriumRegion.startInd]
+        start_psi = self.equilibriumRegion.psi(*start_point)
+
+        # set sign of step in psi towards this region from primary separatrix at start of region
+        if temp_psi_vals[-1] - start_psi > 0:
+            start_psi_sep_plus_delta = start_psi + self.user_options.poloidal_spacing_delta_psi
+        else:
+            start_psi_sep_plus_delta = start_psi - self.user_options.poloidal_spacing_delta_psi
+
+        # set sign of step in psi towards this region from primary separatrix at end of region
+        if temp_psi_vals[-1] - end_psi > 0:
+            end_psi_sep_plus_delta = end_psi + self.user_options.poloidal_spacing_delta_psi
+        else:
+            end_psi_sep_plus_delta = end_psi - self.user_options.poloidal_spacing_delta_psi
```

Comment:
Sorry, my fault I think, but presumably this needs to be after `end_psi` is defined on l. 318 below.

Reply:
Thanks! Sorry I checked without pulling all changes from github. Now working